administrator/components/com_fields/models/fields/type.php

```diff
@@ -91,4 +107,31 @@ protected function getOptions ()
 
 		return $options;
 	}
+
+	/**
+	 * Parses the file with the given path. If it is a class starting with the
+	 * name JFormField and implementing JFormDomfieldinterface, then the class name is returned.
+	 *
+	 * @param string $path
+	 * @return string|boolean
+	 */
+	private function getClassNameFromFile($path)
+	{
+		$tokens = token_get_all(JFile::read($path));
```

Comment:
As I said, the instanceof approach is what I was doing first (and would be the normal approach). But in this case it is different. The JFormFieldDatabaseConnection would meet then both requirements, instanceof JFormField and JFormDomfieldinterface. But it should not be part of the available field types. So how would you solve that then?